runtime/near-vm-runner/src/prepare/prepare_v3.rs

```diff
@@ -514,6 +535,17 @@ mod test {
                 return Err(PrepareError::TooManyLocals);
             }
         }
+        // Similarly, do the same for the number of tables.
+        if table_count.ok_or(PrepareError::TooManyTables)? > MAX_TABLES_PER_MODULE {
+            return Err(PrepareError::TooManyTables);
+        }
+        // Similarly, do the same for the number of table elements.
+        if usize::try_from(max_elements_per_table)
+            .map_err(|_| PrepareError::TooManyTableElements)?
+            > MAX_ELEMENTS_PER_TABLE
+        {
+            return Err(PrepareError::TooManyTableElements);
+        }
```

Comment:
These are going to need gating by a protocol version/feature. Two ways to go about it: either follow the `near_parameters::vm::Config::reftypes_bulk_memory` field for the setup example, add a new boolean field along the lines of `introduce_table_limits` and enable that field in protocol version 81. `LimitConfig::max_locals_per_contract` is another way to approach it wherein the maximums would be configured numerically via parameters, rather than constants.

Reply:
The reason this is a protocol change despite no contract actually using more than 1 table (and the relevant number of elements) is because I have no way to check if anybody actually tried to deploy such contracts in the past. So making it gated by a protocol version is simply a safe way to go about it. I haven't thought about this before...